sycl/include/CL/sycl/detail/device_impl.hpp

```diff
@@ -21,44 +20,141 @@ namespace sycl {
 class platform;
 
 namespace detail {
-// TODO: 4.6.4 Partitioning into multiple SYCL devices
-// TODO: 4.6.4.2 Device information descriptors
+
+// Forward declaration
+class platform_impl;
+class platform_impl_pi;
+
 // TODO: Make code thread-safe
 class device_impl {
 public:
-  virtual ~device_impl() = default;
```

Comment:
It is not clear to me why you remove this `virtual` while you have classes that inherit from and that you use though some shared pointers... Probably it works be cause the shared pointer copy the destructor for the final type...
You do no longer use dynamic polymorphism?

Reply:
@keryell there were only two classes that inherited from device_impl: device_impl_pi and device_host. There are no such classes anymore and no need for virtual destructor.

Reply:
It will be complex then if we want to add completely different runtime (non using the PI story) but this is fine.